src/tests/test_nd_verifier.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/nd/nd.h"
+#include "algorithms/nd/nd_verifier/nd_verifier.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+
+namespace tests {
+namespace onam = config::names;
+
+struct NDVerifyingParams {
+    algos::StdParamsMap params;
+
+    NDVerifyingParams(config::IndicesType lhs_indices, config::IndicesType rhs_indices,
+                      model::WeightType weight = -1, CSVConfig const& csv_config = kTestND,
```

Comment:
`model::WeightType` is `unsigned`. So you should be careful with `-1`.

If you are looking to get the maximum value of a certain type, it is better to use the std::numeric_limits class:

```cpp
#include <limits>
T max = std::numeric_limits<T>::max()
```

You can also use `static_cast<unsigned>(-1)`.